xbmc/addons/kodi-dev-kit/include/kodi/AddonBase.h

```diff
@@ -75,6 +75,94 @@ struct ATTR_DLL_LOCAL CPrivateBase
   static AddonGlobalInterface* m_interface;
 };
 
+/*!
+ * @brief Internally used helper to dynamically allocate and fill a char array.
+ */
+inline char* ATTR_DLL_LOCAL AllocAndCopyString(const char* source)
+{
+  if (source)
+  {
+    const size_t len{strlen(source)};
+    char* target = new char[len + 1];
+    strncpy(target, source, len);
+    target[len] = '\0';
```

Comment:
Less code and a little faster :slightly_smiling_face: 
```suggestion
    memcpy(target, source, len+1);
```